sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobCheckpointStore.java

```diff
@@ -107,71 +110,119 @@ public BlobCheckpointStore(BlobContainerAsyncClient blobContainerAsyncClient, Cl
+                                .log(Messages.NO_METADATA_AVAILABLE_FOR_BLOB);
+                        return;
+                    }
+
+                    final T converted = converter.apply(blobItem, names);
+
+                    if (!Objects.isNull(converted)) {
+                        sink.next(converted);
+                    }
+                });
     }
 
-    private Mono<Checkpoint> convertToCheckpoint(BlobItem blobItem) {
-        String[] names = blobItem.getName().split(BLOB_PATH_SEPARATOR);
-        LOGGER.atVerbose()
-            .addKeyValue(BLOB_NAME_LOG_KEY, blobItem.getName())
-            .log(Messages.FOUND_BLOB_FOR_PARTITION);
-        if (names.length == 5) {
-            // Blob names should be of the pattern
-            // fullyqualifiednamespace/eventhub/consumergroup/checkpoints/<partitionId>
-            // While we can further check if the partition id is numeric, it may not necessarily be the case in future.
-
-            if (CoreUtils.isNullOrEmpty(blobItem.getMetadata())) {
-                LOGGER.atWarning()
-                    .addKeyValue(BLOB_NAME_LOG_KEY, blobItem.getName())
-                    .log(Messages.NO_METADATA_AVAILABLE_FOR_BLOB);
-                return Mono.empty();
-            }
-
-            Map<String, String> metadata = blobItem.getMetadata();
-            LOGGER.atVerbose()
-                .addKeyValue(BLOB_NAME_LOG_KEY, blobItem.getName())
-                .addKeyValue(SEQUENCE_NUMBER_LOG_KEY, metadata.get(SEQUENCE_NUMBER))
-                .addKeyValue(OFFSET_LOG_KEY, metadata.get(OFFSET))
-                .log(Messages.CHECKPOINT_INFO);
-
-            Long sequenceNumber = null;
-            Long offset = null;
-            if (!CoreUtils.isNullOrEmpty(metadata.get(SEQUENCE_NUMBER))) {
-                sequenceNumber = Long.parseLong(metadata.get(SEQUENCE_NUMBER));
-            }
-
-            if (!CoreUtils.isNullOrEmpty(metadata.get(OFFSET))) {
-                offset = Long.parseLong(metadata.get(OFFSET));
-            }
+    /**
+     * Converts a blob item and its path parts into a checkpoint.
+     *
+     * @param blobItem Storage blob item.
+     * @param names Blob path split into parts.
+     * @return The checkpoint.—
+     */
+    private static Checkpoint convertToCheckpoint(BlobItem blobItem, String[] names) {
+        final Map<String, String> metadata = blobItem.getMetadata();
 
-            Checkpoint checkpoint = new Checkpoint()
-                .setFullyQualifiedNamespace(names[0])
-                .setEventHubName(names[1])
-                .setConsumerGroup(names[2])
-                // names[3] is "checkpoint"
-                .setPartitionId(names[4])
-                .setSequenceNumber(sequenceNumber)
-                .setOffset(offset);
+        Long sequenceNumber = null;
+        Long offset = null;
+        if (!CoreUtils.isNullOrEmpty(metadata.get(SEQUENCE_NUMBER))) {
```

Comment:
nit: Don't know how much a hot path this is but we could reduce the number of Map.gets by using an external variable to store while doing null or empty checks:

```suggestion
        String sequenceNumber = metadata.get(SEQUENCE_NUMBER);
        if (!CoreUtils.isNullOrEmpty(sequenceNumber)) {
```